### PR TITLE
feat(general): add policy name and guidelines to CSV output

### PR DESCRIPTION
**Pull request by mikeurbanski1**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Adds the policy name and guidelines (URL or custom policy guidelines) to the CSV output.

I added them as columns at the end so that it doesn't break any automations that users may use that relies on the column order.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by kartikp10 — APPROVED** [review]

File: checkov/common/output/csv.py [hunk 1]
```diff
@@ -39,7 +39,7 @@
 FILE_NAME_CONTAINER_IMAGES = f"{date_now}_container_images.csv"
 
 FILE_NAME_IAC = f"{date_now}_iac.csv"
-HEADER_IAC = ["Resource", "Path", "Git Org", "Git Repository", "Misconfigurations", "Severity"]
+HEADER_IAC = ["Resource", "Path", "Git Org", "Git Repository", "Misconfigurations", "Severity", "Policy name", "Guidelines"]
```
Comment (try-panwiac) [inline checkov/common/output/csv.py:42]:
```suggestion
HEADER_IAC = ["Resource", "Path", "Git Org", "Git Repository", "Misconfigurations", "Severity", "Policy title", "Guideline"]
```

File: checkov/common/output/csv.py [hunk 2]
```diff
@@ -131,6 +135,8 @@ def add_iac_resources(self, resource: Record | ExtraResource, git_org: str, git_
             "Git Repository": git_repository,
             "Misconfigurations": misconfig,
             "Severity": severity,
+            "Policy name": check_name,
```
Comment (try-panwiac) [inline checkov/common/output/csv.py:138]:
```suggestion
            "Policy title": check_name,
```

File: checkov/common/output/csv.py [hunk 3]
```diff
@@ -131,6 +135,8 @@ def add_iac_resources(self, resource: Record | ExtraResource, git_org: str, git_
...
+            "Guidelines": guideline
```
Comment (try-panwiac) [inline checkov/common/output/csv.py:139]:
```suggestion
            "Guideline": guideline
```

**Review by try-panwiac — APPROVED** [review]
Suggestion to align to platform a little

**Review by tsmithv11 — APPROVED** [review]

**Pull request merged**